inbox/api/ns_api.py

```diff
@@ -863,26 +883,43 @@ def event_update_api(public_id):
         # We're going to save this data into a JSON-like TEXT field in the
         # db. With MySQL, this means that the column will be 64k.
         # Drop the latest participants until it fits in the column.
-        while len(json.dumps(cancelled_participants)) > 63000:
+        while len(json.dumps(cancelled_participants)) > 2**15:
             cancelled_participants.pop()
 
     # Don't update an event if we don't need to.
     if noop_event_update(event, data):
         return g.encoder.jsonify(event)
 
-    for attr in Event.API_MODIFIABLE_FIELDS:
-        if attr in data:
-            setattr(event, attr, data[attr])
+    if g.api_version == 1:
+        # Optimistic update!
+        for attr in Event.API_MODIFIABLE_FIELDS:
+            if attr in data:
+                setattr(event, attr, data[attr])
 
-    event.sequence_number += 1
-    g.db_session.commit()
+        event.sequence_number += 1
+        g.db_session.commit()
+
+        # Don't sync back updates to autoimported events.
+        # FIXME @karim: document this and forbid it.
```

Comment:
Rather than having this just silently not do anything can we go ahead and return a 400 with a helpful error message? I'd rather people discover the limitations of our API while using it than discover that it sometimes doesn't work for an unknown reason.